commit 27005 版本编译失败，编译wol-0.7.1的时候发生错误

反馈bug/问题模板，提建议请删除

## 1.关于你要提交的问题

Q：是否搜索了issue (使用 "x" 选择)
* [x] 没有类似的issue

## 2. 详细叙述

编译wol模块的时候发生编译错误

### (1) 具体问题

wol编译失败

### (2) 路由器型号和固件版本

A：x86

### (3) 详细日志

A：

Makefile:240: recipe for target 'getpass4.o' failed
make[6]: Leaving directory '/home/runner/work/lede/lede/build_dir/target-x86_64_musl/wol-0.7.1/lib'
Makefile:230: recipe for target 'all-recursive' failed
make[5]: Leaving directory '/home/runner/work/lede/lede/build_dir/target-x86_64_musl/wol-0.7.1'
Makefile:183: recipe for target 'all' failed
make[4]: Leaving directory '/home/runner/work/lede/lede/build_dir/target-x86_64_musl/wol-0.7.1'
Makefile:50: recipe for target '/home/runner/work/lede/lede/build_dir/target-x86_64_musl/wol-0.7.1/.built' failed
make[3]: Leaving directory '/home/runner/work/lede/lede/package/lean/wol'
time: package/lean/wol/compile#0.14#0.05#0.19
package/Makefile:111: recipe for target 'package/lean/wol/compile' failed
make[2]: Leaving directory '/home/runner/work/lede/lede'
package/Makefile:107: recipe for target '/home/runner/work/lede/lede/staging_dir/target-x86_64_musl/stamp/.package_compile' failed
make[1]: Leaving directory '/home/runner/work/lede/lede'
/home/runner/work/lede/lede/include/toplevel.mk:225: recipe for target 'world' failed
In file included from /home/runner/work/lede/lede/staging_dir/toolchain-x86_64_gcc-7.5.0_musl/include/fortify/stdio.h:22:0,
                 from getpass4.c:3:
/home/runner/work/lede/lede/staging_dir/toolchain-x86_64_gcc-7.5.0_musl/include/stdio.h:160:9: note: previous declaration of 'getdelim' was here
 ssize_t getdelim(char **__restrict, size_t *__restrict, int, FILE *__restrict);
         ^~~~~~~~
make[6]: *** [getpass4.o] Error 1
make[5]: *** [all-recursive] Error 1
make[4]: *** [all] Error 2
make[3]: *** [/home/runner/work/lede/lede/build_dir/target-x86_64_musl/wol-0.7.1/.built] Error 2
make[2]: *** [package/lean/wol/compile] Error 2
make[1]: *** [/home/runner/work/lede/lede/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make: *** [world] Error 2
##[error]Process completed with exit code 2.
 Prepare artifact

--- a/package/lean/wol/patches/001-fix-build-on-x86.patch
+++ b/package/lean/wol/patches/001-fix-build-on-x86.patch
@@ -1,0 +1,19 @@
+diff --git a/lib/getline.h b/lib/getline.h
+index 991184c..e41865c 100644
+--- a/lib/getline.h
++++ b/lib/getline.h
+@@ -27,6 +27,7 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+ #  endif
+ # endif
+ 
++# ifdef _MUSL_
+ # if __GLIBC__ < 2
+ int
+ getline PARAMS ((char **_lineptr, size_t *_n, FILE *_stream));
+@@ -34,5 +35,6 @@ getline PARAMS ((char **_lineptr, size_t *_n, FILE *_stream));
+ int
+ getdelim PARAMS ((char **_lineptr, size_t *_n, int _delimiter, FILE *_stream));
+ # endif
++# endif
+ 
+ #endif /* not GETLINE_H_ */

